state/watcher_profilecharm_test.go

```diff
@@ -303,6 +318,7 @@ func noop() {
 // getting information to private channels.
 type channelMatcher struct {
 	changes []watcher.Change
+	done    chan struct{}
 }
 
 func (m channelMatcher) Matches(x interface{}) bool {
```

Comment:
It looks like you are using this just to perform side-effects. The validation of type isn't a strong behaviour verification.

Why not ditch the matcher, allow `gomock.Any()` and do the cast and channel send in a `.Do(...)`?